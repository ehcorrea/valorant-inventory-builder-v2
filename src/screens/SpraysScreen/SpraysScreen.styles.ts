import { ButtonSpray } from '@/components';
import styled from 'styled-components/native';

export const Container = styled.View`
  padding-horizontal: 8px;
`;

export const ContainerRow = styled.View`
  flex-direction: row;
  flex: 1;
  justify-content: center;
  width: 100%;
`;

export const Spray = styled(ButtonSpray)`
  margin: 8px;
`;
