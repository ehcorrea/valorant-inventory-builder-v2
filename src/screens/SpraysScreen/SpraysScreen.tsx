import { ScrollView } from 'react-native';
import { router } from 'expo-router';

import { spraysStore } from '@/stores';

import * as S from './SpraysScreen.styles';

export function SpraysScreen() {
  const { sprays } = spraysStore();

  const handlePushSkinsSprays = (replace: number) => {
    router.push({ pathname: '/skins/sprays', params: { replace } });
  };

  return (
    <ScrollView bounces={false}>
      <S.Container>
        <S.ContainerRow>
          <S.Spray
            accessibilityLabel="select spray 1"
            image={sprays[0].fullIcon}
            onPress={() => handlePushSkinsSprays(0)}
          />
          <S.Spray
            accessibilityLabel="select spray 2"
            image={sprays[1].fullIcon}
            onPress={() => handlePushSkinsSprays(1)}
          />
        </S.ContainerRow>
        <S.ContainerRow>
          <S.Spray
            accessibilityLabel="select spray 3"
            image={sprays[2].fullIcon}
            onPress={() => handlePushSkinsSprays(2)}
          />
          <S.Spray
            accessibilityLabel="select spray 4"
            image={sprays[3].fullIcon}
            onPress={() => handlePushSkinsSprays(3)}
          />
        </S.ContainerRow>
      </S.Container>
    </ScrollView>
  );
}
