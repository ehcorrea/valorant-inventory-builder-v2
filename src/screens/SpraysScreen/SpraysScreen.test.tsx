import { fireEvent, mockedRouter, render } from '@/test/utils';

import { SpraysScreen } from './SpraysScreen';

const setup = () => render(<SpraysScreen />);

describe('<SpraysScreen/>', () => {
  describe('when press a spray button', () => {
    test.each([0, 1, 2, 3])('given that it is in position %s', (position) => {
      const container = setup();
      const buttonSpray = container.getByLabelText(
        `select spray ${position + 1}`
      );

      if (!position) {
        expect(container).toMatchSnapshot();
      }

      fireEvent.press(buttonSpray);
      expect(mockedRouter.push).toHaveBeenCalledWith({
        params: { replace: position },
        pathname: '/skins/sprays',
      });
    });
  });
});
