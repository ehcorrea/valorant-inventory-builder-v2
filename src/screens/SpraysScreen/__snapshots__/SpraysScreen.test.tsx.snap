// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpraysScreen/> when press a spray button given that it is in position 0 1`] = `
<RCTScrollView
  bounces={false}
>
  <View>
    <View
      style={
        {
          "paddingHorizontal": 8,
        }
      }
    >
      <View
        style={
          {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityLabel="select spray 1"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "aspectRatio": 1,
              "backgroundColor": "#cfcfcf11",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 8,
              "marginLeft": 8,
              "marginRight": 8,
              "marginTop": 8,
              "maxHeight": 250,
              "maxWidth": 250,
              "opacity": 1,
            }
          }
        >
          <Image
            contentFit="cover"
            source={1}
            style={
              {
                "height": "90%",
                "width": "90%",
              }
            }
          />
        </View>
        <View
          accessibilityLabel="select spray 2"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "aspectRatio": 1,
              "backgroundColor": "#cfcfcf11",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 8,
              "marginLeft": 8,
              "marginRight": 8,
              "marginTop": 8,
              "maxHeight": 250,
              "maxWidth": 250,
              "opacity": 1,
            }
          }
        >
          <Image
            contentFit="cover"
            source={1}
            style={
              {
                "height": "90%",
                "width": "90%",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityLabel="select spray 3"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "aspectRatio": 1,
              "backgroundColor": "#cfcfcf11",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 8,
              "marginLeft": 8,
              "marginRight": 8,
              "marginTop": 8,
              "maxHeight": 250,
              "maxWidth": 250,
              "opacity": 1,
            }
          }
        >
          <Image
            contentFit="cover"
            source={1}
            style={
              {
                "height": "90%",
                "width": "90%",
              }
            }
          />
        </View>
        <View
          accessibilityLabel="select spray 4"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "aspectRatio": 1,
              "backgroundColor": "#cfcfcf11",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 8,
              "marginLeft": 8,
              "marginRight": 8,
              "marginTop": 8,
              "maxHeight": 250,
              "maxWidth": 250,
              "opacity": 1,
            }
          }
        >
          <Image
            contentFit="cover"
            source={1}
            style={
              {
                "height": "90%",
                "width": "90%",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
